.github/workflows/pr-diff.yml

```diff
@@ -0,0 +1,82 @@
+name: PR Diff
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  issues: write
+  pull-requests: write
+  contents: read
+
+jobs:
+  pr_diff:
+    runs-on: ubuntu-latest
+
+    # Only run when a comment containing `/prdiff` is created
+    # and the author is a member, collaborator or owner
+    if: >
+      (
+        github.event_name == 'issue_comment' &&
+        github.event.issue.pull_request &&
+        contains(fromJSON('["MEMBER", "COLLABORATOR", "OWNER"]'), github.event.comment.author_association) &&
+        startsWith(github.event.comment.body, '/prdiff')
+      )
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Read comment
+      id: read_comment
+      env:
+        COMMENT_BODY: ${{ github.event.comment.body }}
+      run: |
+        if [[ "$COMMENT_BODY" =~ ^/prdiff[[:space:]]([^[:space:]]+) ]]; then
+          CMD_ARG=${BASH_REMATCH[1]}
+          echo "Using cmd argument: $CMD_ARG"
+          echo "other_pr=$CMD_ARG" >> $GITHUB_OUTPUT
```

Comment:
Will there be any risk of command injections? Like

```
/prdiff https://github.com/xxx `do something evil`
```

This may not be a practical example, but we should be extra careful when processing untrusted input (such as these from comments). 

@curiositycasualty could you help to double check here as well?

Reply:
@dndx I just had a similar discussion with @ADD-SP, it should not be a major concern since we limit execution of this workflow to members, owners, maintainers: https://github.com/Kong/kong/pull/13401/files#diff-84cd7992011f4a1f7fe5735c840973ddeecc24d45d25a978b5f1c1709ef7707cR16-R24

Still, we decided it'd be a good idea to validate the input. I just added a step that checks that we are only passing a valid and expected URL https://github.com/Kong/kong/pull/13401/files#diff-84cd7992011f4a1f7fe5735c840973ddeecc24d45d25a978b5f1c1709ef7707cR43-R61

Reply:
Leveraging `env:` to lean on bash/shell escaping is what mitigates injection in actions workflows. So this looks as-good-as-can-be to me.

Reply:
Marking this as resolved given @curiositycasualty 's comment.